providers/google/src/airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -1091,14 +1091,14 @@ def generate_query(self, hook: BigQueryHook) -> str:
 
     def execute(self, context: Context):
         if self.project_id:
-            self.log.warning(
+            warnings.warn(
                 "The project_id parameter is deprecated, and will be removed in a future release."
-                " Please use table_project_id instead.",
+                " Please use table_project_id instead.",AirflowDeprecationWarning
```

Comment:
Google Provider has it's own deprecation mechanism with depraction date - see https://github.com/apache/airflow/blob/main/providers/google/docs/deprecation-policy.rst - cc: @VladaZakharova